api/sms_provider/vonage.go

```diff
@@ -0,0 +1,78 @@
+package sms_provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/netlify/gotrue/conf"
+)
+
+const (
+	defaultVonageApiBase = "https://rest.nexmo.com"
+)
+
+type VonageProvider struct {
+	Config  *conf.VonageProviderConfiguration
+	APIPath string
+}
+
+type VonageResponseMessage struct {
+	Status    string `json:"status"`
+	ErrorText string `json:"error-text"`
+}
+
+type VonageResponse struct {
+	Messages []VonageResponseMessage `json:"messages"`
+}
+
+// Creates a SmsProvider with the Vonage Config
+func NewVonageProvider(config conf.VonageProviderConfiguration) (SmsProvider, error) {
+	if err := config.Validate(); err != nil {
+		return nil, err
+	}
+
+	apiPath := defaultVonageApiBase + "/sms/json"
+	return &VonageProvider{
+		Config:  &config,
+		APIPath: apiPath,
+	}, nil
+}
+
+// Send an SMS containing the OTP with Vonage's API
+func (t VonageProvider) SendSms(phone string, message string) error {
+	body := url.Values{
+		"from":       {t.Config.From},
+		"to":         {phone},
```

Comment:
does vonage require the phone number to be prefixed with a "+" ? by default, we strip out the "+" when we validate the phone number.

Reply:
Vonage doesn't require the number to be prefixed with a "+". I tried this implementation and I got the SMS successfully.

Reply:
You actually don't even need a phone number. In many countries, you can use an Alphanumeric SenderID of up to 11 characters (8 for India). However, in some more restrictive countries this would need to be registered.
Full list of supported countries with Alphanumeric Sender ID can be found here: https://help.nexmo.com/hc/en-us/articles/115011781468-Outbound-SMS-Features-Overview